inception-recommendation-api/src/main/java/de/tudarmstadt/ukp/inception/recommendation/api/model/AnnotationObject.java

```diff
@@ -18,239 +18,202 @@
+        return confidence;
     }
 
-    public void setId(int id) {
-        this.id = id;
+    public void setConfidence(double aConfidence)
+    {
+        confidence = aConfidence;
     }
-    
-    public void setRecommenderId(long recommendationId)
+
+    public long getRecommenderId()
     {
-        this.recommenderId = recommendationId;
-    }  
-    
-    public long getRecommenderId() {
         return recommenderId;
     }
-    
-    @Override
-    public String toString()
+
+    public void setRecommenderId(long aRecommenderId)
     {
-        return "Annotation: " + annotation + " - CoveredText: " + coveredText + " - Offset: "
-                + offset.toString() + " - DocumentURI: " + documentURI + " - DocumentName: " 
-                + documentName +  " - ID: " + id;
+        recommenderId = aRecommenderId;
     }
 
-    @Override
-    public int hashCode()
+    public TokenObject getTokenObject()
     {
-        final int prime = 31;
-        int result = super.hashCode();
-        result = prime * result + ((annotation == null) ? 0 : annotation.hashCode());
-        result = prime * result + ((sentenceTokens == null) ? 0 : sentenceTokens.hashCode());
-        return result;
+        return token;
+    }
+
+    public String getDocumentName()
+    {
+        return token.documentName;
     }
 
     @Override
-    public boolean equals(Object obj)
+    public boolean equals(Object o)
     {
-        if (this == obj) {
+        if (this == o)
             return true;
-        }
-        if (!super.equals(obj)) {
-            return false;
-        }
-        if (getClass() != obj.getClass()) {
-            return false;
-        }
-        AnnotationObject other = (AnnotationObject) obj;
-        if (annotation == null) {
-            if (other.annotation != null) {
-                return false;
-            }
-        }
-        else if (!annotation.equals(other.annotation)) {
+        if (o == null || getClass() != o.getClass())
             return false;
-        }
-        if (sentenceTokens == null) {
-            if (other.sentenceTokens != null) {
-                return false;
-            }
-        }
-        else if (!sentenceTokens.equals(other.sentenceTokens)) {
-            return false;
-        }
-        return true;
+        AnnotationObject that = (AnnotationObject) o;
+        return id == that.id && recommenderId == that.recommenderId;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(id, recommenderId);
     }
 
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("AnnotationObject{");
+        sb.append("token=").append(token);
+        sb.append(", id=").append(id);
+        sb.append(", label='").append(label).append('\'');
+        sb.append(", uiLabel='").append(uiLabel).append('\'');
+        sb.append(", feature='").append(feature).append('\'');
+        sb.append(", source='").append(source).append('\'');
```

Comment:
source and recommenderId are kind of redundant - but we can address this later.

Reply:
Isn't the source used for the UI?

Reply:
There are some AnnotationObject constructors which set the source (classifier) to null. A good measure might be to leave it at null and to add throw an exception in `getSource()` if the source has not been set. Then we get an idea where this might be needed but is not provided properly.